src/extension/android/logCatMonitor.ts

```diff
@@ -0,0 +1,84 @@
+   the .vscode/launch.json file by defining a setting named logCatArguments for the configuration being used. The
+   setting accepts values as:
+      1. an array: ["*:S", "ReactNative:V", "ReactNativeJS:V"]
+      2. a string: "*:S ReactNative:V ReactNativeJS:V"
+   Type `adb logcat --help` to see the parameters and usage of logcat
+*/
+export class LogCatMonitor implements vscode.Disposable {
+    private static DEFAULT_PARAMETERS = ["*:S", "ReactNative:V", "ReactNativeJS:V"];
+
+    private _childProcess: ChildProcess;
+    private _logger = new OutputChannelLogger(vscode.window.createOutputChannel("LogCat"));
+
+    private _deviceId: string;
+    private _userProvidedLogCatArguments: any; // This is user input, we don't know what's here
+
+    private _logCatSpawn: ISpawnResult;
+
+    constructor(deviceId: string, userProvidedLogCatArguments: string, { childProcess = new ChildProcess() } = {}) {
+        this._deviceId = deviceId;
+        this._userProvidedLogCatArguments = userProvidedLogCatArguments;
+        this._childProcess = childProcess;
+    }
+
+    public start(): Q.Promise<void> {
+        const logCatArguments = this.logCatArguments();
+        const adbParameters = ["-s", this._deviceId, "logcat"].concat(logCatArguments);
+        this._logger.logMessage(`Monitoring LogCat for device ${this._deviceId} with arguments: ${logCatArguments}`);
+
+        this._logCatSpawn = new ChildProcess().spawnWithExitHandler("adb", adbParameters);
+
+        /* LogCat has a buffer and prints old messages when first called. To ignore them,
+            we won't print messages for the first 0.5 seconds */
+        const filter = new ExecutionsFilterBeforeTimestamp(/*delayInSeconds*/ 0.5);
+        this._logCatSpawn.stderr.on("data", (data: Buffer) => {
+            filter.execute(() => this._logger.logMessage(data.toString(), /*formatMessage*/ false));
+        });
+
+        this._logCatSpawn.stdout.on("data", (data: Buffer) => {
+            filter.execute(() => this._logger.logMessage(data.toString(), /*formatMessage*/ false));
+        });
+
+        return this._logCatSpawn.outcome.then(
+            () =>
+                this._logger.logMessage("LogCat monitoring stopped because the process exited."),
+            reason => {
+                if (!this._logCatSpawn) { // We stopped log cat ourselves
+                    this._logger.logMessage("LogCat monitoring stopped because the debugging session finished");
+                } else {
+                    return Q.reject<void>(reason); // Unkown error. Pass it up the promise chain
+                }
+            }).finally(() =>
+                this._logCatSpawn = null);
+    }
+
+    public dispose(): void {
+        if (this._logCatSpawn) {
+            const logCatSpawn = this._logCatSpawn;
+            this._logCatSpawn = null;
+            logCatSpawn.spawnedProcess.kill();
```

Comment:
I tested it on my machine (Win10) and it worked. I'm not sure if it might break on a different version of windows.